src/main/java/org/pf4j/update/PluginInfo.java

```diff
@@ -76,9 +77,9 @@ public PluginRelease getLastRelease(Version systemVersion) {
      * @param installedVersion version that is already installed
      * @return true if there is a newer version available which is compatible with system
      */
-    public boolean hasUpdate(Version systemVersion, Version installedVersion) {
+    public boolean hasUpdate(String systemVersion, String installedVersion) {
         PluginRelease last = getLastRelease(systemVersion);
-        return last != null && Version.valueOf(last.version).greaterThan(installedVersion);
+        return last != null && Version.valueOf(last.version).greaterThan(Version.valueOf(installedVersion));
```

Comment:
Here is the real problem. Maybe a new method, `greaterThan`, somewhere (`UpdateManager` for example) or use a comparator.